devito/operator/operator.py

```diff
@@ -663,8 +647,11 @@ def _soname(self):
 
     @cached_property
     def ccode(self):
-        from devito.ir.iet.visitors import CGen
-        return CGen(compiler=self._compiler).visit(self)
+        try:
+            return self._ccode_handler(compiler=self._compiler).visit(self)
```

Comment:
So this is the door for custom operators in pro/... right? Wouldn't making `CGen` the default instead of `None` avoid this try block?

Reply:
it catches both an AttributeError and a TypeError but I don't remember exactly why, to be fair. My bad. I will see if I can improve it